File: backend/app/schemas/registry.py
from datetime import datetime

from pydantic import BaseModel

# --------------------------------------------------------------------------
# PLACEHOLDER CONTRACT — Agent Templates and SkillHub Registry. A real sync
# integration for these two sources lives elsewhere and will be wired in
# later. This schema + the endpoints in app/api/v1/endpoints/registry.py
# define the response *shape* the frontend is built against now;
# app/crud/registry.py's in-memory stub is what to replace with the real
# sync logic later — the route paths, request/response shapes, and
# admin-only gating should stay as-is so the frontend doesn't need to change
# when the real backend lands.
#
# MCP and Model started out as placeholder sources here too, but were
# replaced by a real availability-sync implementation directly on the mcps/
# ai_models tables (see app/api/v1/endpoints/mcps.py, ai_models.py) rather
# than staying a second, fake "mirrored registry" source of truth.
# --------------------------------------------------------------------------


class RegistryItem(BaseModel):
    id: str
    name: str
    version: str | None
    category: str | None
    deprecated: bool
    last_seen_at: datetime | None


class RegistryStatus(BaseModel):
    total_count: int
    last_synced_at: datetime | None
    consecutive_failures: int
    stale_count: int


class RegistryOverview(BaseModel):
    status: RegistryStatus
    items: list[RegistryItem]

File: frontend/src/api/registry.ts
import { apiClient } from "./client";
import type { RegistryOverview, RegistrySource, RegistryStatus } from "./types";

// See backend/app/schemas/registry.py's PLACEHOLDER note: these four sources are
// currently backed by an in-memory stub pending the real sync integration (owned
// elsewhere). The contract below is what that integration will fill in — this file
// shouldn't need to change when it lands.
export async function getRegistryOverview(source: RegistrySource): Promise<RegistryOverview> {
  const { data } = await apiClient.get<RegistryOverview>(`/admin/${source}`);
  return data;
}

export async function triggerResync(source: RegistrySource): Promise<RegistryStatus> {
  const { data } = await apiClient.post<RegistryStatus>(`/admin/${source}/resync`);
  return data;
}
